### PR TITLE
Update ffmpeg to 5.1.2

### PR DESCRIPTION
**Pull request by omartijn**

### CODE REVIEW

File: ports/ffmpeg/1000-resolume-small-memory-allocations.patch [hunk 1]
```diff
@@ -56,7 +56,7 @@ index 858633e8c7..b96eab0a7b 100644
  
 -    ret = av_buffer_create(data, size, av_buffer_default_free, NULL, 0);
 -    if (!ret)
--        av_freep(&data);
+         av_freep(&data);
```
Comment (sandercox) [inline ports/ffmpeg/1000-resolume-small-memory-allocations.patch]: I think this is an error and should not be there.

File: ports/ffmpeg/1003-avformat_index_get_entry_const_correctness.patch [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+-const AVIndexEntry *avformat_index_get_entry_from_timestamp(AVStream *st,
++const AVIndexEntry *avformat_index_get_entry_from_timestamp(const AVStream *st,
+                                                             int64_t wanted_timestamp,
+                                                             int flags)
+ {
+-    const FFStream *const sti = ffstream(st);
++    const FFStream *const sti = cffstream(st);
+     int idx = ff_index_search_timestamp(sti->index_entries,
+                                         sti->nb_index_entries,
+                                         wanted_timestamp, flags);
```
Comment (sandercox) [inline ports/ffmpeg/1003-avformat_index_get_entry_const_correctness.patch:64]: Did you file a patch for that with FFmpeg too as its obviously wrong?

File: ports/ffmpeg/1004-codec-info.patch [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+ }
+ 
++int av_stream_get_codec_info_nb_frames(const AVStream *st)
++{
++    return cffstream(st)->codec_info_nb_frames;
++}
++
+ int avformat_query_codec(const AVOutputFormat *ofmt, enum AVCodecID codec_id,
+                          int std_compliance)
+ {
```
Comment (sandercox) [inline ports/ffmpeg/1004-codec-info.patch]: We should not use these fields if they are not in the public API - should not add these 'to get it to compile'

**Review by sandercox — CHANGES_REQUESTED** [review]
It's hard to review as we don't compare what is from vcpkg upstream and our changes.

But I managed to do that locally.

* There is a bug in the custom allocator.
* We should mail in 1003 on the public ffmpeg mailing list I guess.
* We should remove patch 1004 and fix anywhere where it it used.

File: ports/ffmpeg/1003-avformat_index_get_entry_const_correctness.patch (hunk 2 above)
Comment (omartijn) [reply ports/ffmpeg/1003-avformat_index_get_entry_const_correctness.patch:64]: I had done that, but looks like that didn't go through correctly. It seems I sent it too quickly after signing up so it wasn't processed. I've re-submitted the patch now.

File: ports/ffmpeg/1000-resolume-small-memory-allocations.patch (hunk 1 above)
Comment (omartijn) [reply ports/ffmpeg/1000-resolume-small-memory-allocations.patch]: You're right! No idea how that got here. Fixed it.

File: ports/ffmpeg/1004-codec-info.patch (hunk 3 above)
Comment (omartijn) [reply ports/ffmpeg/1004-codec-info.patch]: Yes. I've removed this patch now.

**Review by sandercox — APPROVED** [review]

**Pull request merged**